crates/holochain_p2p/tests/tests/node_messaging.rs

```diff
@@ -164,21 +164,74 @@ async fn test_call_remote() {
     let dna_hash = DnaHash::from_raw_36(vec![0; 36]);
     let handler = Arc::new(Handler::default());
 
-    let (agent1, _hc1) = spawn_test(dna_hash.clone(), handler.clone()).await;
-    let (_agent2, hc2) = spawn_test(dna_hash.clone(), handler).await;
+    let (agent1, hc1) = spawn_test(dna_hash.clone(), handler.clone()).await;
+    let (agent2, hc2) = spawn_test(dna_hash.clone(), handler).await;
+
+    tokio::time::timeout(std::time::Duration::from_secs(5), async {
+        loop {
+            tokio::time::sleep(std::time::Duration::from_millis(1)).await;
+
+            // make sure h2 has its own address
+            if hc2
+                .peer_store(dna_hash.clone())
+                .await
+                .unwrap()
+                .get_all()
+                .await
+                .unwrap()
+                .len()
+                > 0
+            {
+                break;
+            }
+        }
+
+        loop {
+            tokio::time::sleep(std::time::Duration::from_millis(1)).await;
+
+            let resp = hc2
+                .call_remote(
+                    dna_hash.clone(),
+                    agent1.clone(),
+                    ExternIO(b"hello".to_vec()),
+                    Signature([0; 64]),
+                )
+                .await
+                .unwrap();
+            let resp: Vec<u8> = UnsafeBytes::from(resp).into();
+            let resp = String::from_utf8_lossy(&resp);
+            assert_eq!("got_call_remote: hello", resp);
+
+            // break when hc1 has hc2's address
+            if hc1
+                .peer_store(dna_hash.clone())
+                .await
+                .unwrap()
+                .get_all()
+                .await
+                .unwrap()
+                .len()
+                > 1
+            {
+                break;
+            }
+        }
+    })
+    .await
+    .unwrap();
 
-    let resp = hc2
+    let resp = hc1
         .call_remote(
```

Comment:
This call previously didn't work, since hc1 was created before hc2 and thus didn't have it from the bootstrapping process.

We've still got some awkward timing (see the loops above) because async tasks are generating the preflight data, but I don't *think* that'll be a problem in the real world.

Reply:
That's a usual case in the real world, isn't it? Where one agent comes online before another and therefore the first hasn't seen the second but the second may already be making a call to the first.

Reply:
In the real world, will the preflight data have been generated synchronously before making a connection?

Reply:
No, not synchronously. Asynchronously. But who are we making a connection to? Both kitsune's gossip algorithm and holochain itself only learn about peers at all asynchronously, So I presume there will be time post-join for this code to run before a connection establishment is requested.

Reply:
The remote call looks up the url from the peer store first. It seems the test would be closer to this if hc1 awaited for hc2 to appear in its peer store first.

Reply:
yes, but. The mem bootstrap runs on a five second loop. i don't want our teats to be that slow... we could make it run more often, but that would make it use more cpu while we're trying to teat other stuff...

Reply:
Alright already! =D